BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -112,6 +119,33 @@ public virtual void Initialize(TelemetryConfiguration configuration, TelemetryMo
             this.Initialize(configuration, modules, PlatformSingleton.Current.ReadConfigurationXml());
         }
 
+        [SuppressMessage("Microsoft.Reliability", "CA2000:DisposeObjectsBeforeLosingScope", Justification = "We want objects created in this method to live for the life of the application.")]
+        [SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", Justification = "This value is not used in NetStandard1.3 but is used for all other frameworks..")]
+        internal void EvaluateDiagnosticsMode(TelemetryModules modules)
+        {
+#if !NETSTANDARD1_3
+            // TODO: NEED TO EMIT ETW LOGS
+            if (PlatformSingleton.Current.TryGetEnvironmentVariable(DiagnosticsEnvironmentVariable, out string diagnosticsValue) && bool.TryParse(diagnosticsValue, out bool boolValue) && boolValue)
+            {
+                if (modules != null && !modules.Modules.Any(module => module is FileDiagnosticsTelemetryModule))
```

Comment:
We may want to make Environment variables (Diag=true/false and LogDir=///) win over configuration file for the same parameters, so that these values can be affected outside of the process, just like connection string variables are. In the current implementation we seem to ignore LogDir variable if FileDiagnosticsTelemetryModule is already configured.

That also brings a question on what to do if FileDiagnostics Module is mentioned in both config and variables - do we keep both module or only register first/discard the second or something else?

Reply:
If i inject a Module via EnVar, AND a Module exists in the customer's config... currently the TCFactory will override the first (EnvVar) with the one from the config.
THIS is a problem because the new module will have lost all of the SDK initialization ETW traces.

Note for self: 
On Monday:
Add a new derived private class  `SelfFileDiagnosticsmodule : FileDiagnosticsModule`.
We will inject the private one via EnVar and this should not conflict with the one from the config file.